docs/registry-authentication.md

```diff
@@ -0,0 +1,26 @@
+# Registry Authentication
+
+Machine-controller supports configuring container-runtime with authentication
+information. Flag `-node-registry-credentials-secret` can take a secret
+reference where authentication info will be stored. During VM creation this info
+will be used to configure container-runtime.
+
+Secret format is serialized `map[string]github.com/containerd/containerd/pkg/cri/config.AuthConfig`
```

Comment:
Similar to @ahmedwaleedmalik’s comment, this is not really intuitive, so we should probably replace it with a link to the struct.